x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/__tests__/get_beats_stats.js

```diff
@@ -0,0 +1,138 @@
+      expect(body.size).to.be(10000);
+    });
+
+    it('should set `from: 10000, from: 10000` in the query', () => {
+      fetchBeatsStats(server, callCluster, clusterUuids, start, end, { page: 1 });
+      const { args } = callCluster.firstCall;
+      const [api, { body }] = args;
+
+      expect(api).to.be('search');
+      expect(body.from).to.be(10000);
+      expect(body.size).to.be(10000);
+    });
+
+    it('should set `from: 20000, from: 10000` in the query', () => {
+      fetchBeatsStats(server, callCluster, clusterUuids, start, end, { page: 2 });
+      const { args } = callCluster.firstCall;
+      const [api, { body }] = args;
+
+      expect(api).to.be('search');
+      expect(body.from).to.be(20000);
+      expect(body.size).to.be(10000);
+    });
+  });
+
+  describe('handleBeatsResultSet', () => {
+    it('should summarize empty results', () => {
+      const expectedEmpty = {};
+
+      const resultsEmpty = handleBeatsResultSet([{}, {}, {}]);
+      expect(resultsEmpty).to.eql(expectedEmpty);
+
+      const resultsQuiteEmpty = handleBeatsResultSet([]);
+      expect(resultsQuiteEmpty).to.eql(expectedEmpty);
+
+      const resultsTotallyEmpty = handleBeatsResultSet();
+      expect(resultsTotallyEmpty).to.eql(expectedEmpty);
+    });
+
+    it('should summarize single result with some missing fields', () => {
+      const results = handleBeatsResultSet([
+        {
+          hits: {
+            hits: [
+              {
+                _source: {
+                  cluster_uuid: 'FlV4ckTxQ0a78hmBkzzc9A',
+                  beats_stats: {
+                    metrics: { libbeat: { output: { type: 'elasticsearch' } } }, // missing events published
+                    beat: { type: 'cowbeat' }, // missing version and output
+                  },
+                },
+              },
+            ],
+          },
+        },
+      ]);
+
+      expect(results).to.eql({
+        FlV4ckTxQ0a78hmBkzzc9A: {
+          count: 1,
+          versions: {},
+          types: { cowbeat: 1 },
+          outputs: { elasticsearch: 1 },
+          eventsPublished: 0,
+          hosts: 0,
+        },
+      });
+    });
+
+    it('should summarize stats from hits across multiple result objects', () => {
+      const results = handleBeatsResultSet(beatsStatsResultSet); // NOTE: HITS_SIZE was 2 when capturing these results, so the set has hits in 205 result objects
+      const expected = {
+        W7hppdX7R229Oy3KQbZrTw: {
+          count: 5,
+          versions: { '7.0.0-alpha1': 5 },
+          types: { metricbeat: 1, filebeat: 4 },
+          outputs: { elasticsearch: 5 },
+          eventsPublished: 80875,
+          hosts: 1,
+        },
+        FlV4ckTxQ0a78hmBkzzc9A: {
+          count: 405,
+          versions: { '7.0.0-alpha1': 405 },
+          types: {
+            filebeat: 200,
+            metricbeat: 100,
+            heartbeat: 100,
+            winlogbeat: 1,
+            duckbeat: 1,
+            'apm-server': 1,
```

Comment:
NOTE: it's possible to filter `apm-server` out of the search, but I figure it's useful for telemetry

Reply:
6d010fb427